chore: Add Prettier auto-formatting Github Action

Rohit3523 opened the pull request:
## Proposed changes
I have implemented a new GitHub Actions workflow to automatically format our codebase using Prettier. This workflow runs automatically on every commit, with the exception of direct pushes to the `develop`, `master` and `single-server` branch in the main repo but will run on every single branch in forked repo (check the image attached below)

The workflow utilizes Node.js 22 (as per the current LTS schedule: https://nodejs.org/en/about/previous-releases#release-schedule) and npm ci for dependable dependency installation. When triggered, it runs the Prettier formatter across the codebase, and if any formatting changes are detected, it automatically commits and pushes these changes with a standardized commit message: "chore: format code with Prettier".

This automation eliminates manual formatting tasks, reduces style-related discussions during code reviews, and maintains a consistent code style throughout our codebase. The workflow has been thoroughly tested across various scenarios including push events to feature branches and pull request events, confirming both its functionality and proper branch exclusion rules.

## Issue(s)	
None

## Example
https://github.com/Test3523/Rocket.Chat.ReactNative/pull/11

## Types of changes
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves a current function)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist
- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

## How this action works
When you open a Pull Request, GitHub Actions will:
1. Run Prettier to format your code.
2. Check for changes:
2.1 ✅ No changes? The action completes.
2.2 🔄 Changes found? A commit is made with message "chore: format code with Prettier [skip ci]"
    
### Detailed explanation when this condition will run
<img width="745" alt="Screenshot 2025-03-06 at 2 15 34 AM" src="https://github.com/user-attachments/assets/08b1e8ad-ce32-412f-a57e-4e3665f31746" />

[Reference for skip ci](https://docs.github.com/en/actions/managing-workflow-runs-and-deployments/managing-workflow-runs/skipping-workflow-runs)

diegolmello commented on .github/workflows/prettier.yml:
```diff
@@ -0,0 +1,56 @@
+
+jobs:
+  format:
+    if: ${{ github.repository != 'RocketChat/Rocket.Chat.ReactNative' }} 
```
Why it doesn't run on this repo?
This is useful for us as well.

Rohit3523 replied:
This action is disabled on this repo because when a user creates or updates a pull request from their fork, the action runs both in the forked repo and the main repo.

- In the forked repo, the action successfully formats and commits changes.
- In this repo, the action also runs but fails because it doesn’t have permission to push changes to the user’s forked branch.

To prevent these failures, we disable this action in the main repo, allowing it to run only in forks where it can successfully push changes.

Rohit3523 replied:
I am updating the action script to run the action here if it's not from a fork repo

diegolmello reviewed (changes requested):
Can you solve my comment and trigger everything again?

Rohit3523 commented:
I have updated the GitHub Actions workflow to run on all branches of forked repositories. In the main Rocket.Chat repository, it will execute on all branches except master, develop, and single-server.

To prevent loops, I have added `[skip ci]` to the commit messages of Prettier commits. This ensures that these specific commits do not trigger the workflow, avoiding potential infinite loops. [Reference](https://docs.github.com/en/actions/managing-workflow-runs-and-deployments/managing-workflow-runs/skipping-workflow-runs)

<img width="751" alt="Screenshot 2025-03-06 at 2 12 38 AM" src="https://github.com/user-attachments/assets/ae4d2f64-0b17-48b1-bb48-e507a94ba39a" />
